### ISSUE TITLE
Add the possibility to use plaintext keys if key file ends with '.txt' extension

### ISSUE THREAD

**Issue by maki-chan** [maki-chan/wudecrypt#2]
For a more convenient use of the program, especially for less experienced users, WUDecrypt should add the possibility to use plaintext key files where the key is not the raw byte representation.

This way, people can just copy & paste their keys into text files without the need of a hex editor or something similar.

**Comment by Draky50110** [comment]
Yes :)

**Comment by Whovian9369** [comment]
Definitely! Making bins manually is quite confusing, haha!

**Comment by mrohnstock** [comment]
as these keys are short SHA1 hashes, entering these by parameter would be great, too.

i.e.:

```
./wudecrypt -c COMMONKEY-HASH -d DISCKEY-HASH -o OUTPUTPATH FILE.wud
```

or something like that.

**Comment by maki-chan** [comment]
@mrohnstock I agree to you that this could be possible. Though, they are not SHA1 hashes. They a raw encryption keys for an AES-CBC encryption (AES-128 = 128 bits = 16 bytes). A SHA1 hash is 20 bytes long.

There's only one problem that makes me thinking about it twice: Windows and Unix systems have different ways of handling command line arguments in a _good_ way. Unix uses the, in my opinion, extremely great GNU `getopt` method, I don't know if Windows has such an alternative.
But in the future, I will also support better parameter handling. I just need to include another library for that, like I already do for SHA1 and dynamic arrays.

**Comment by Whovian9369** [comment]
If we wanted to try doing this ourselves, what file should we start in? (Curiosity strikes me)

**Comment by maki-chan** [comment]
You would need to add a function header to `functions.h`, add the function itself (the implementation) to `functions.c` and then you would need to check for the `.txt` file extension in the `main.c` file in the `main` method to see if we need to load the key from a file directly or if we need to load the key from a plaintext file via the new method you added in `functions.h` and `functions.c`.

Though that's just one possibility and the way I would go. You could also add the check and load algorithm directly to the already existing `loadKey` and `loadKeyFile` methods.

The code isn't very clean at the moment, so it doesn't really matter where you add it.

**Comment by Whovian9369** [comment]
Just wanted to ask, in case I somehow learned C before you added it :P Thanks!

**Comment by mdeguzis** [comment]
yes please! :) I want to break away from using Uwizard (Windows only) with Wine, and use this exclusively. I only keep my title keys etc. in plain text files. I'm kinda new to the scene anyway. Not sure how to make my .bin files.
